samples: failing to build due to rebase

**Is your enhancement proposal related to a problem? Please describe.**
After rebasing to a more recent Zephyr commit, the `samples/lib/thrift/hello_client` and `samples/lib/thrift/hello_server` applications are failing to build with the error
```
/__w/gsoc-2022-thrift/gsoc-2022-thrift/thrift-for-zephyr/samples/lib/thrift/hello_client/src/main.cpp:28:1: error: '::main' must return 'int'
   28 | void main(void)
```

Additionally, after fixing that, I've seen the following when building for e.g. `mps2_an521`, `qemu_cortex_a53`:
```
ccache /home/cfriedt/zephyr-sdk-0.14.2/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc -DKERNEL -DMBEDTLS_CONFIG_FILE=\"config-tls-generic.h\" -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/cfriedt/my-workspace/zephyr/include/zephyr -I/home/cfriedt/my-workspace/zephyr/include -I/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated -I/home/cfriedt/my-workspace/zephyr/soc/arm64/qemu_cortex_a53 -I/home/cfriedt/my-workspace/zephyr/lib/libc/newlib/include -I/home/cfriedt/my-workspace/zephyr/include/zephyr/posix -I/home/cfriedt/my-workspace/zephyr/lib/util/fnmatch/. -I/home/cfriedt/my-workspace/zephyr/subsys/net/l2 -I/home/cfriedt/my-workspace/zephyr/subsys/net/lib/sockets/. -I/home/cfriedt/my-workspace/zephyr/subsys/net/lib/conn_mgr/. -I/home/cfriedt/my-workspace/thrift-for-zephyr/lib/thrift/include -I/home/cfriedt/my-workspace/thrift-for-zephyr/lib/thrift/../../.upstream/lib/cpp/src -I/home/cfriedt/my-workspace/modules/crypto/mbedtls/include -I/home/cfriedt/my-workspace/zephyr/modules/mbedtls/configs -Os -imacros /home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-a53 -mabi=lp64 --sysroot=/home/cfriedt/zephyr-sdk-0.14.2/aarch64-zephyr-elf/aarch64-zephyr-elf -imacros /home/cfriedt/my-workspace/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/cfriedt/my-workspace/thrift-for-zephyr/samples/lib/thrift/hello_server=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/cfriedt/my-workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/cfriedt/my-workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -D_SYS__PTHREADTYPES_H_ -DMISSING_SYSCALL_NAMES -D_POSIX_TIMERS -D_POSIX_MONOTONIC_CLOCK -std=c99 -MD -MT zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_pipe.c.obj -MF zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_pipe.c.obj.d -o zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_pipe.c.obj -c /home/cfriedt/my-workspace/zephyr/drivers/serial/uart_pipe.c
In file included from /home/cfriedt/my-workspace/zephyr/include/zephyr/toolchain/gcc.h:88,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/toolchain.h:50,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/sys/cbprintf.h:13,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_msg.h:11,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log.h:11,
                 from /home/cfriedt/my-workspace/zephyr/drivers/serial/uart_pipe.c:14:
/home/cfriedt/my-workspace/zephyr/drivers/serial/uart_pipe.c: In function 'uart_pipe_register':
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:96:39: error: '__device_dts_ord_DT_CHOSEN_zephyr_uart_pipe_ORD' undeclared (first use in this function)
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:96:31: note: in expansion of macro '_CONCAT'
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:276:37: note: in expansion of macro 'DEVICE_NAME_GET'
  276 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:296:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  296 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/cfriedt/my-workspace/zephyr/drivers/serial/uart_pipe.c:103:18: note: in expansion of macro 'DEVICE_DT_GET'
  103 |  uart_pipe_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_uart_pipe));
      |                  ^~~~~~~~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:96:39: note: each undeclared identifier is reported only once for each function it appears in
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:96:31: note: in expansion of macro '_CONCAT'
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:276:37: note: in expansion of macro 'DEVICE_NAME_GET'
  276 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/cfriedt/my-workspace/zephyr/include/zephyr/device.h:296:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  296 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/cfriedt/my-workspace/zephyr/drivers/serial/uart_pipe.c:103:18: note: in expansion of macro 'DEVICE_DT_GET'
  103 |  uart_pipe_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_uart_pipe));
      |                  ^~~~~~~~~~~~~
[145/314] Building C object zephyr/subsys/net/lib/config/CMakeFiles/subsys__net__lib__config.dir/init.c.obj
In file included from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_msg.h:663,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log.h:11,
                 from /home/cfriedt/my-workspace/zephyr/subsys/net/lib/config/init.c:9:
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h: In function 'z_log_msg_static_create.constprop':
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h:24:20: note: parameter passing for argument of type 'const struct log_msg_desc' changed in GCC 9.1
   24 | static inline void z_log_msg_static_create(const void * source, const struct log_msg_desc desc, uint8_t * package, const void * data)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
[166/314] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/net_if.c.obj
In file included from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_msg.h:663,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log.h:11,
                 from /home/cfriedt/my-workspace/zephyr/subsys/net/ip/net_if.c:7:
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h: In function 'z_log_msg_static_create.constprop':
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h:24:20: note: parameter passing for argument of type 'const struct log_msg_desc' changed in GCC 9.1
   24 | static inline void z_log_msg_static_create(const void * source, const struct log_msg_desc desc, uint8_t * package, const void * data)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
[167/314] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp.c.obj
In file included from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_msg.h:663,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log.h:11,
                 from /home/cfriedt/my-workspace/zephyr/subsys/net/ip/tcp.c:7:
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h: In function 'z_log_msg_static_create.constprop':
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h:24:20: note: parameter passing for argument of type 'const struct log_msg_desc' changed in GCC 9.1
   24 | static inline void z_log_msg_static_create(const void * source, const struct log_msg_desc desc, uint8_t * package, const void * data)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
[168/314] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
In file included from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_msg.h:663,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/cfriedt/my-workspace/zephyr/include/zephyr/logging/log.h:11,
                 from /home/cfriedt/my-workspace/zephyr/subsys/net/ip/ipv6_nbr.c:16:
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h: In function 'z_log_msg_static_create.constprop':
/home/cfriedt/my-workspace/thrift-for-zephyr/build/zephyr/include/generated/syscalls/log_msg.h:24:20: note: parameter passing for argument of type 'const struct log_msg_desc' changed in GCC 9.1
   24 | static inline void z_log_msg_static_create(const void * source, const struct log_msg_desc desc, uint8_t * package, const void * data)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.

```

**Describe the solution you'd like**
Samples should build without error.

**Describe alternatives you've considered**


**Additional context**

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -51,7 +51,7 @@ jobs:
         shell: bash
         run: |
           source zephyr-env.sh
-          twister -i --build-only -W -T samples/
+          twister -i --build-only -G -T samples/
 
       - name: Build Samples (POSIX)
         working-directory: thrift-for-zephyr

--- a/samples/lib/thrift/hello_client/sample.yaml
+++ b/samples/lib/thrift/hello_client/sample.yaml
@@ -2,22 +2,18 @@ sample:
   description: Hello Thrift sample
   name: hello thrift
 common:
-  tags: samples thrift cpp
+  tags: thrift cpp
   # FIXME: zephyrproject-rtos/zephyr#45100
   arch_exclude: posix
+  filter: TOOLCHAIN_HAS_NEWLIB == 1
+  toolchain_exclude: xcc
+  arch_exclude: arc xtensa
+  integration_platforms:
+    - mps2_an385
+    - qemu_x86_64
 tests:
   thrift.hello.newlib:
-    filter: TOOLCHAIN_HAS_NEWLIB == 1
-    toolchain_exclude: xcc
-    platform_exclude: qemu_x86 qemu_leon3 qemu_malta qemu_malta_be
-    arch_exclude: arc xtensa
     tags: newlib
 #  thrift.hello.native:
 #    arch_allow: posix
 #    tags: native
-  thrift.hello.arcmwdtlib:
-    toolchain_allow: arcmwdt
-    platform_allow: nsim_hs nsim_em
-    tags: arc
-    extra_configs:
-      - CONFIG_ARCMWDT_LIBC=y

--- a/samples/lib/thrift/hello_client/src/main.cpp
+++ b/samples/lib/thrift/hello_client/src/main.cpp
@@ -25,7 +25,7 @@ using namespace apache::thrift::protocol;
 using namespace apache::thrift::transport;
 
 #ifdef __ZEPHYR__
-void main(void)
+int main(void)
 #else
 int main(int argc, char** argv)
 #endif
@@ -59,12 +59,8 @@ int main(int argc, char** argv)
     transport->close();
   } catch (std::exception& e) {
     printf("caught exception: %s\n", e.what());
-#ifndef __ZEPHYR__
     return EXIT_FAILURE;
-#endif
   }
 
-#ifndef __ZEPHYR__
   return EXIT_SUCCESS;
-#endif
 }

--- a/samples/lib/thrift/hello_server/sample.yaml
+++ b/samples/lib/thrift/hello_server/sample.yaml
@@ -2,22 +2,18 @@ sample:
   description: Hello Thrift sample
   name: hello thrift
 common:
-  tags: samples thrift cpp
+  tags: thrift cpp
   # FIXME: zephyrproject-rtos/zephyr#45100
   arch_exclude: posix
+  filter: TOOLCHAIN_HAS_NEWLIB == 1
+  toolchain_exclude: xcc
+  arch_exclude: arc xtensa
+  integration_platforms:
+    - mps2_an385
+    - qemu_x86_64
 tests:
   thrift.hello.newlib:
-    filter: TOOLCHAIN_HAS_NEWLIB == 1
-    toolchain_exclude: xcc
-    platform_exclude: qemu_x86 qemu_leon3 qemu_malta qemu_malta_be
-    arch_exclude: arc xtensa
     tags: newlib
 #  thrift.hello.native:
 #    arch_allow: posix
 #    tags: native
-  thrift.hello.arcmwdtlib:
-    toolchain_allow: arcmwdt
-    platform_allow: nsim_hs nsim_em
-    tags: arc
-    extra_configs:
-      - CONFIG_ARCMWDT_LIBC=y

--- a/samples/lib/thrift/hello_server/src/main.cpp
+++ b/samples/lib/thrift/hello_server/src/main.cpp
@@ -25,7 +25,7 @@ using namespace ::apache::thrift::transport;
 using namespace ::apache::thrift::server;
 
 #ifdef __ZEPHYR__
-void main(void)
+int main(void)
 #else
 int main(int argc, char** argv)
 #endif
@@ -55,12 +55,8 @@ int main(int argc, char** argv)
     server.serve();
   } catch (std::exception& e) {
     printf("caught exception: %s\n", e.what());
-#ifndef __ZEPHYR__
     return EXIT_FAILURE;
-#endif
   }
 
-#ifndef __ZEPHYR__
   return EXIT_SUCCESS;
-#endif
 }

